Missing stdbool.h include in zephyr/kernel/stats.h

**Describe the bug**
It seems that a header kernel file misses include. Include of `<stdbool.h>` or` <zephyr/types.h>`, as bool type is used inside `k_cycle_stats` struct. Upgrading from Zephyr 3.0.99 to 3.2.99 introduced this issue in my project, as Zephyr won't compile anymore as this is treated as error. 

**To Reproduce:**
Not sure what configuration is causing this as it did, but this happens by basically invoking `west build` for custom target board. Issue was not occurring with Zephyr 3.0.99

**Expected behavior:**
Zephyr compiles without errors. 

**Impact:**
Build system generates error because it does not recognize "bool" type. 

**Logs and console output:**

```
In file included from C:\work\git\kona\zephyr\include\zephyr\kernel\thread.h:14,
                 from c:\work\git\projects\my_project\src\main.c:7:
C:\work\git\projects\zephyr\include\zephyr\kernel\stats.h:24:9: error: unknown type name 'bool'
   24 |         bool      track_usage;  /* true if gathering usage stats */
      |         ^~~~
C:\work\git\projects\zephyr\include\zephyr\kernel\thread.h:35:9: error: unknown type name 'k_thread_entry_t'
   35 |         k_thread_entry_t pEntry;
      |         ^~~~~~~~~~~~~~~~
C:\work\git\projects\zephyr\include\zephyr\kernel\thread.h:50:9: error: unknown type name 'sys_dnode_t'
   50 |         sys_dnode_t      node;
      |         ^~~~~~~~~~~
C:\work\git\projects\zephyr\include\zephyr\kernel\thread.h:61:17: error: unknown type name 'sys_dnode_t'
   61 |                 sys_dnode_t qnode_dlist;
      |                 ^~~~~~~~~~~
C:\work\git\projects\zephyr\include\zephyr\kernel\thread.h:62:31: error: field 'qnode_rb' has incomplete type
   62 |                 struct rbnode qnode_rb;
      |                               ^~~~~~~~
C:\work\git\projects\zephyr\include\zephyr\kernel\thread.h:68:9: error: unknown type name '_wait_q_t'
   68 |         _wait_q_t *pended_on;
      |         ^~~~~~~~~
C:\work\git\projects\zephyr\include\zephyr\kernel\thread.h:131:25: error: field 'timeout' has incomplete type
  131 |         struct _timeout timeout;
      |                         ^~~~~~~
C:\work\git\projects\zephyr\include\zephyr\kernel\thread.h:237:9: error: unknown type name 'bool'
  237 |         bool is_polling;
      |         ^~~~
C:\work\git\projects\zephyr\include\zephyr\kernel\thread.h:250:30: error: field 'callee_saved' has incomplete type
  250 |         struct _callee_saved callee_saved;
      |                              ^~~~~~~~~~~~
C:\work\git\projects\zephyr\include\zephyr\kernel\thread.h:256:9: error: unknown type name '_wait_q_t'
  256 |         _wait_q_t join_queue;
      |         ^~~~~~~~~
C:\work\git\projects\zephyr\include\zephyr\kernel\thread.h:343:29: error: field 'arch' has incomplete type
  343 |         struct _thread_arch arch;
```

**Environment:**
 - Windows
 - Zephyr SDK 0.15.2
 - Zephyr 3.2.99 version

**Additional context:**

--- a/include/zephyr/kernel/stats.h
+++ b/include/zephyr/kernel/stats.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2021, Intel Corporation
+ * Copyright (c) 2021,2023, Intel Corporation
  *
  * SPDX-License-Identifier: Apache-2.0
  */
@@ -8,6 +8,7 @@
 #define ZEPHYR_INCLUDE_KERNEL_STATS_H_
 
 #include <stdint.h>
+#include <stdbool.h>
 
 /*
  * [k_cycle_stats] is used to track internal statistics about both thread

